### PR TITLE
Add recursemode parameter to chose to apply ACL's recursivel

### PR DESCRIPTION
**Pull request by roidelapluie**
This commit adds a Recursemode parameter to generate additional
resources for children files when recurse is set to true.

Depends on dobbymoodge/puppet-acl#11.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (dobbymoodge) [inline .travis.yml]: @roidelapluie Shouldn't this file be in a separate commit? This is part of the Travis support for test running, not part of the recursive ACL changes, right?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
+Checklist (and a short version for the impatient)
```
Comment (dobbymoodge) [inline CONTRIBUTING.md]: @roidelapluie This file should be part of a separate PR, too?

**Comment by dobbymoodge** [comment]
@roidelapluie Sorry for being so slow to merge these. This looks good, except all the test support stuff - nodesets, `.travis.yml`, etc. - need to be put in a separate pull request unless they relate directly to the ACL recursion changes. That is, it makes sense for [spec/unit/puppet/type/acl_spec.rb](spec/unit/puppet/type/acl_spec.rb) to be in this pull request, but a lot of the other stuff should be in a separate PR so the history is a bit clearer.

I should have reviewed this before I merged in #11, since that would have been the right place for most of these files. Oh well. :)

**Comment by dobbymoodge** [comment]
Also, maybe update `README.org` to document the new options (`:lazy`, `:deep`)

**Comment by dobbymoodge** [comment]
But what the heck, this project is so low-traffic, having the PR do double-duty isn't such a big deal. :)

**Comment by roidelapluie** [comment]
rebased and noted your comment. Should we silently disable default ACL's for files?

**Comment by dobbymoodge** [comment]
That's what `setfacl -R` does, so I think we should do the same.

**Comment by dobbymoodge** [comment]
Changes from this PR pulled in with additional fixes by dobbymoodge/puppet-acl#13
